core/arch/arm/plat-imx/drivers/imx_caam.c

```diff
@@ -9,9 +9,148 @@
 #include <initcall.h>
 #include <io.h>
 #include <mm/core_memprot.h>
+#include <imx.h>
+#include <imx-regs.h>
+#include <io.h>
+#include <kernel/generic_boot.h>
+#include <kernel/tee_common_otp.h>
+#include <mm/core_memprot.h>
+#include <platform_config.h>
+#include <stdint.h>
+#include <tee/cache.h>
 
 #include "imx_caam.h"
 
+static uint8_t stored_key[MKVB_SIZE];
+static bool mkvb_retrieved;
+
+static void caam_enable_clocks(void)
+{
+	vaddr_t ccm_base = core_mmu_get_va(CCM_BASE, MEM_AREA_IO_SEC);
+
+	io_setbits32(ccm_base + CCM_CCGR0,
+		     CCM_CCGR0_CAAM_WRAPPER_IPG  |
+		     CCM_CCGR0_CAAM_WRAPPER_ACLK |
+		     CCM_CCGR0_CAAM_SECURE_MEM);
+
+	if (soc_is_imx6dqp() || soc_is_imx6sdl() || soc_is_imx6dq())
+		io_setbits32(ccm_base + CCM_CCGR6,
+			     CCM_CCGR6_EMI_SLOW);
+}
+
+static TEE_Result imx_caam_reset_jr(struct imx_caam_ctrl *ctrl)
+{
+	uint32_t reg_val = 0;
+	uint32_t timeout = 1000;
+
+	io_write32((vaddr_t)&ctrl->jrcfg[MKVB_JR].jrcr, 1);
```

Comment:
It would be nice that raw value (`1` here) at lines 50, 55, 59, ... are replaced with macros, i.e. `CAAM_JRCR_xxx BIT(0)`, `CAAM_JRINTR_STATE_MASK  GENMASK_32(3, 2)`, ...